app/Http/Controllers/Service/RegistrationController.php

```diff
@@ -26,9 +33,66 @@ public function index()
     /**
      * Stores an incoming Registration.
      *
-     * @param Request $request
+     * @param StoreNewRegistrationRequest $request
+     * @return \Illuminate\Http\RedirectResponse
      */
-    public function store(Request $request)
+    public function store(StoreNewRegistrationRequest $request)
     {
+        // Duplicate families are fine at this point.
+        $family = new Family(['rvid' => Family::generateRVID()]);
+
+        // Create Carers
+        // TODO: Alter request to pre-join the array?
+        $carers = array_map(
+            function ($carer) {
+                return new Carer(['name' => $carer]);
+            },
+            array_merge(
+                (array)$request->get('carer'),
+                (array)$request->get('carers')
+            )
+        );
+
+        // Create Children
+        $children = array_map(
+            function ($child) {
+                // Note: Carbon uses different time formats than laravel validation
+                // Also, format() uses the current day of month if unspecified, so we startOfMonth() it
+                return new Child([
+                        'dob' => Carbon::createFromFormat('Y-m', $child)
+                            ->startOfMonth()
+                            ->format('Y-m-d'),
+                    ]);
+            },
+            (array)$request->get('children')
```

Comment:
Do you need to cast these to array?

Reply:
Yes, array_merge() doesnt seem to auto cast single strings. if there is only one "children" element in the formdata then it comes through as a string rather than a single element array.